.github/actions/build-push-pr.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+cat <<EOF > local.mk
+PUBLISH_URL=https://api.synocommunity.com
+PUBLISH_API_KEY=
+MAINTAINER?=
+MAINTAINER_URL=
+DISTRIBUTOR=SynoCommunity
+DISTRIBUTOR_URL=https://synocommunity.com/
+REPORT_URL=https://github.com/SynoCommunity/spksrc/issues
+SUPPORT_URL=https://github.com/SynoCommunity/spksrc/issues
+DEFAULT_TC=6.1
+PARALLEL_MAKE=4
+EOF
+
+
+# get dependency tree
+DEPENDENCY_TREE=
+echo "Building dependency tree..."
+for package in $(find spk/ -maxdepth 1 -type d | cut -c 5- | sort)
+do
+    pushd "spk/$package" > /dev/null || exit
+    DEPENDENCY_TREE+=$(make dependency-tree)$'\n'
+    popd > /dev/null || exit
+done
+
+# filter for changes made in the cross and spk directories
+GH_FILES=$(echo "$GH_FILES" | grep -oE "(spk.*)|(cross.*)|(native.*)")
+
+# create array of potential packages where files have changed
+GH_PACKAGES_ARR=()
+for file in $GH_FILES
+do
+    # remove leading spk/cross/native from string
+    file=${file#spk/}
+    file=${file#cross/}
+    file=${file#native/}
+    # get package name / folder name
+    package=$(echo "$file" | grep -oE "^[^\/]*")
+    echo "===> Searching for dependent package: $package"
+    packages=$(echo "$DEPENDENCY_TREE" \
+        | awk -v package="$package" \
+        'NF == 2 {x=$2} $2 == package {print x}' \
+         | sort -u)
```

Comment:
This awk command is a bit fragile as it depends that the output columns from `make dependency-tree` does not change, if someone with more awk wizardry can come up with something better please share 👍

Reply:
Does it make sense to use #3854 code?

Reply:
I've tried it, but it failed to return any packages when I modified a single line in a package in cross. I don't know how it works so I assume the best way is to walk the dependency tree.

Reply:
@publicarray you're saying that there's an error in #3854.  Let's fix it.  What's your repro of the error?